source/localizable/tutorial/subroutes.md

```diff
@@ -0,0 +1,398 @@
+  return { data: selectedRental };
+}
+```
+
+Our `show` sub-route only expects to show one rental.
+This is why we will use `queryRecord` instead of `query` as we expect a single response.
+This means that we cannot return an array (like we do with a normal get response to our Mirage file).
+We search all of our rentals until we find the one that matches the slug passed in our `queryParams`.
+
+Since our API is now returning two additional fields, we need to update our `rentals` model, so we can access them in our application.
+
+```app/models/rental.js
+export default DS.Model.extend({
+  title: DS.attr(),
+  owner: DS.attr(),
+  city: DS.attr(),
+  type: DS.attr(),
+  image: DS.attr(),
+  bedrooms: DS.attr(),
+  slug: DS.attr(),
+  description: DS.attr()
+});
+```
+
+# Generating a Show Route
+
+Now that our API is ready to return single records, we can generate our `show` sub-route.
+Much like generating our `rentals` route, we will use `ember g` to create a nested route.
+
+```shell
+ember g route rentals/show
+```
+
+You will see output like this:
+
+```shell
+installing route
+  create app/routes/rentals/show.js
+  create app/templates/rentals/show.hbs
+updating router
+  add route rentals/show
+installing route-test
+  create tests/unit/routes/rentals/show-test.js
+```
+
+Let's start by looking at the changes to our Router (`app/router.js`).
+
+```app/router.js
+Router.map(function() {
+  this.route('about');
+  this.route('contact');
+  this.route('rentals', function() {
+    this.route('show');
+  });
+});
+```
+
+You will notice that `this.route('show')` is nested within our `rentals` route.
+This tells Ember that it is a sub-route and must be accessed through `localhost:4200/rentals/show`.
+
+In order for us to tell the application which rental we want to access, we need to pass the `slug` to our `show` route.
+In addition, we will want to simplify the URL for the user so that they can access the information for a specific rental by browsing to `localhost:4200/rentals/slug-for-rental`.
+To do this, we modify our router.
+
+```app/router.js
+Router.map(function() {
+  this.route('about');
+  this.route('contact');
+  this.route('rentals', function() {
+    this.route('show', { path: '/:slug' });
+  });
+});
+```
+
+We have modified the `path` for the `show` route.
+It no longer defaults to `/rentals/show` but instead `/rentals/:slug`.
+This means we can now access the `slug` on the route itself.
+
+# Querying by Slug
+
+Next, we will want to edit our `show` route to return the specific rental we want.
+
+```app/routes/rentals/show.js
+export default Ember.Route.extend({
+  model(params) {
+    return this.store.queryRecord('rental', { slug: params.slug });
+  }
+});
+```
+
+Since we added `:slug` to the `show` path in our router, we can now access `slug` through the `params` in our `model` hook.
+When we call `this.store.queryRecord('rental', { slug: params.slug })`, Ember Data will make a GET request to `/rentals?slug=our-slug`.
```

Comment:
For consistency, we've been accessing store with a get, as we do other services, such as `this.get('store').queryRecord(...)`